### PR TITLE
Carthage support

### PR DESCRIPTION
**Pull request by jakubvano**
I've made minimal changes to make integration via Carthage possible. 

However, it might need further optimisations as I have not analysed lib structure in detail.

### CODE REVIEW

File: LayoutTest.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -677,6 +677,7 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				ENABLE_BITCODE = NO;
```
Comment (plivesey) [inline LayoutTest.xcodeproj/project.pbxproj]: Why is this necessary?

File: LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (plivesey) [inline LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Is this a new file? Or something which wasn't checked in before?
Why is it needed?

File: LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTest.xcscheme [hunk 3]
```diff
@@ -28,16 +28,6 @@
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
       <Testables>
-         <TestableReference
```
Comment (plivesey) [inline LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTest.xcscheme]: What does this change? And why is it needed?

File: LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (plivesey) [inline LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme]: Presumably this is just making the schema 'shared'?
Why does carthage need this?

File: LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestTests.xcscheme [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (plivesey) [inline LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestTests.xcscheme]: It looks like you've added LayoutTestBase and LayoutTestTests. Why not LayoutTest? And why do you need to make a test schema public?

**Comment by plivesey** [comment]
Thanks for the pull request.

I haven't used Carthage before and am not familiar with how it works. I've left a bunch of comments on the diff which may help me understand, but in general, why are project file changes necessary for Carthage?

It seems like Carthage expects a framework to be built from source. Is that correct?

File: LayoutTest.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jakubvano) [inline LayoutTest.xcodeproj/project.pbxproj]: With bitcode enabled, linking against `XCTest.framework` would fail because it does not contain bitcode.`

File: LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme (hunk 4 above)
Comment (jakubvano) [inline LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme]: Carthage is building only shared framework schemes

**Comment by jakubvano** [comment]
Upon further investigation, creating separate scheme for tests proved unnecessary.

In general: unlike CocoaPods which creates it's own workspace, project and build targets, Carthage is using build schemes provided by the project being built. Thus project settings directly impact carthage build process.

File: LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 2 above)
Comment (jakubvano) [inline LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: This file was generated by Xcode when I opened the project. From what I understand it is not essential, but should be checked in or added to `.gitignore` - common practice is to check it in,

File: LayoutTest.xcodeproj/project.pbxproj (hunk 1 above)
Comment (plivesey) [inline LayoutTest.xcodeproj/project.pbxproj]: Sounds good. Does this change both LayoutTestBase and LayoutTest to be bitcode disabled?

File: LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 2 above)
Comment (plivesey) [inline LayoutTest.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Probably it's in my global ignore list or something...I agree we should check this in.

File: LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme (hunk 4 above)
Comment (plivesey) [inline LayoutTest.xcodeproj/xcshareddata/xcschemes/LayoutTestBase.xcscheme]: Cool.

**Comment by plivesey** [comment]
How can I locally test the changes?

Also, you don't need to add this to a shared carthage repo like cocoapods right? It links directly to the github?

Second question, does carthage rely on tags for versioning? Do you need me to do a release with these changes?

**Comment by jakubvano** [comment]
> How can I locally test the changes?

Provided you have carthage installed, you can test it by creating `Cartfile` with contents

```
github "jakubvano/LayoutTest-iOS" "master"
```

and running `carthage update --platform iOS`. It should create create folder `Carthage/Build/iOS/`containing `LayoutTest.framework` and `LayoutTestBase.framework`.

>Also, you don't need to add this to a shared carthage repo like cocoapods right? It links directly to the github?

Correct, it links directly to the github.

>Second question, does carthage rely on tags for versioning? Do you need me to do a release with these changes?

In `Cartfile` one can specify exact branch / tag / commit to use with carthage, by default it checks out tag with latest version. Ideally there should be release containing proposed changes, so that people can use 

```
github "linkedin/LayoutTest-iOS"
```
in their `Cartfile`

File: LayoutTest.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jakubvano) [inline LayoutTest.xcodeproj/project.pbxproj]: Yes, this disables it for both

**Comment by plivesey** [comment]
Thanks. I'll test this today and then merge it into master.

**Comment by plivesey** [comment]
Looks like everything is working. One last question though before merging:

There are two frameworks. I presume that Carthage will always download both, even if I just use one and that that is the accepted practice for stuff like this? Is that correct?

**Comment by plivesey** [comment]
Regardless of the answer to the above question, I think this change looks good. If we want to make future changes we can.

Merging...
